contracts/Sacd.sol

```diff
@@ -362,6 +403,35 @@ contract Sacd is ISacd, Initializable, AccessControlUpgradeable, UUPSUpgradeable
     paymentId = _getSacdStorage().nextPaymentId[asset][grantee][grantor];
   }
 
+  /**
+   * @notice Checks if a template is active
+   * @dev Internal function to check template status
+   * @param templateId The ID of the template to check
+   * @return bool Returns true if the template is active
+   */
+  function _isTemplateActive(uint256 templateId) internal view returns (bool) {
+    // Early return for most common case (no template used)
+    if (templateId == 0) return true;
+
+    SacdStorage storage $ = _getSacdStorage();
+    address templateContract = $.templateContract;
+
+    // Early return if no template contract set
+    if (templateContract == address(0)) return true;
+
+    // Check if template exists by trying to get its owner
+    try ITemplate(templateContract).ownerOf(templateId) returns (address) {
+      // If owner exists, check if template is active
+      try ITemplate(templateContract).isTemplateActive(templateId) returns (bool isActive) {
```

Comment:
Do we expect this to revert? Does `isTemplateActive` not take care of all of the cases from 423–432?